lib/cadet/accounts/ivle.ex

```diff
@@ -107,7 +109,60 @@ defmodule Cadet.Accounts.IVLE do
     end
   end
 
-  defp api_fetch(method, queries) do
+  @doc """
+  Make an API call to IVLE LAPI.
+
+  returns...
+
+    - {:ok, body} - valid token
+    - {:error, :internal_server_error} - Invalid API key
+    - {:error, :bad_request} - Invalid token
+
+  ## Parameters
+
+    - method: String, the HTTP request method to use
+    - queries: [Keyword], key-value pair of parameters to send
+
+  This method is valid for methods that return with a string "Invalid login!"
+  in a JSON nested in the body. Refer to the next method `api_call/2` for
+  methods that return a 200 with an empty string body on invalid tokens.
+  """
+  def api_call(method, queries) when method in ["Announcements"] do
+    with {:ok, %{status_code: 200, body: body}} <- HTTPoison.get(api_url(method, queries)),
+         body = Poison.decode!(body),
+         %{"Comments" => "Valid login!"} <- body do
+      {:ok, body["Results"]}
+    else
+      {:ok, %{status_code: 500}} ->
+        # IVLE responds with 500 if APIKey is invalid
+        {:error, :internal_server_error}
+
+      %{"Comments" => "Invalid login!"} ->
+        # IVLE response if AuthToken is invalid
+        {:error, :bad_request}
+    end
+  end
+
+  @doc """
+  Make an API call to IVLE LAPI.
+
+  returns...
+
+    - {:ok, body} - valid token
+    - {:error, :internal_server_error} - Invalid API key
+    - {:error, :bad_request} - Invalid token
+
+  ## Parameters
+
+    - method: String, the HTTP request method to use
+    - queries: [Keyword], key-value pair of parameters to send
+
+  This method is valid for methods that return a 200 with an empty string body
+  on invalid tokens. For methods that return with string "Invalid login!" in a
+  JSON nested in the body, refer to the previous method api_call/2 with guard
+  clause.
+  """
+  def api_call(method, queries) do
     case HTTPoison.get(api_url(method, queries)) do
       {:ok, %{body: body, status_code: 200}} when body != ~s("") ->
```

Comment:
Instead of doing
```elixir
{:ok, %{body: body, status_code: 200}} when body != ~s("") ->
{:ok, %{body: ~s(""), status_code: 200}} ->
```
Maybe you could consider doing 
```elixir
{:ok, %{body: ~s(""), status_code: 200}} ->
{:ok, %{body: body, status_code: 200}} ->
```

Since `case` pattern matching is evaluated sequentially

Reply:
I prefer the current match order, since it follows an 'if ok, else error' structure, rather than an 'if error, elif ok, else error'.